dynawo/sources/Modeler/DataInterface/DYNComponentInterface.cpp

```diff
@@ -111,6 +124,30 @@ ComponentInterface::getType() const {
   return type_;
 }
 
+const std::string&
+ComponentInterface::getTypeAsString() const {
+  static const std::string stringTypes[] = {
+      "UNKNOWN",
+      "BUS",
+      "CALCULATED_BUS",
+      "SWITCH",
+      "LOAD",
+      "LINE",
+      "GENERATOR",
+      "SHUNT",
+      "DANGLING_LINE",
+      "TWO_WTFO",
+      "THREE_WTFO",
+      "SVC",
+      "VSC_CONVERTER",
+      "LCC_CONVERTER",
+      "HVDC_LINE"
+    };
+  ComponentType_t type = type_;
+  if (type < UNKNOWN || type > HVDC_LINE)
```

Comment:
Please add an additional type to the enum (named MAX_TYPE?) to replace this HVDC_LINE that could easily change position in the enum and introduce silent bugs

Reply:
Fixed